src/registers/model_specific.rs

```diff
@@ -161,6 +181,43 @@ bitflags! {
     }
 }
 
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Clone, Copy)]
+/// Memory types used in the [PAT](Pat).
+#[repr(u8)]
+pub enum PatMemoryType {
+    /// Disables caching.
+    Uncacheable = 0x00,
+    /// Uses a write combining cache policy.
+    WriteCombining = 0x01,
+    /// Uses a write through cache policy.
+    WriteThrough = 0x04,
+    /// Uses a write protected cache policy.
+    WriteProtected = 0x05,
+    /// Uses a write back cache policy.
+    WriteBack = 0x06,
+    /// Same as uncacheable, but can be overridden by MTRRs.
+    Uncached = 0x07,
```

Comment:
I'm a little concerned that the names `Uncacheable` and `Uncached` are too similar to each other. Intel's manual refers to these as "Strong Uncacheable" and "Uncacheable" respectively. IMHO these names are less confusing.